haystack/graph_retriever/text_to_sparql.py

```diff
@@ -0,0 +1,93 @@
+import logging
+from typing import Optional
+
+from haystack.graph_retriever.base import BaseGraphRetriever
+
+from transformers import BartForConditionalGeneration, BartTokenizer
+
+logger = logging.getLogger(__name__)
+
+
+class Text2SparqlRetriever(BaseGraphRetriever):
+    """
+        Graph retriever that uses a pre-trained Bart model to translate natural language questions given in text form to queries in SPARQL format.
+        The generated SPARQL query is executed on a knowledge graph.
+    """
+
+    def __init__(self, knowledge_graph, model_name_or_path, top_k: int = 1):
+        self.knowledge_graph = knowledge_graph
+        self.model = BartForConditionalGeneration.from_pretrained(model_name_or_path, force_bos_token_to_be_generated=True)
```

Comment:
We should extend this in the next iteration to any seq2seq models and use the AutoModel class

Reply:
Okay, added a todo note